docs/solutions/rhel74_image/ipxe.sh.template

```diff
@@ -0,0 +1,161 @@
+
+# DO NOT CHANGE - PRE-GENERATED VALUES #
+SSH_PUBLIC_KEY=<PUBLIC_KEY>
+TENANCY=<TENANCY>
+USER=<USER>
+FINGERPRINT=<FINGERPRINT>
+REGION=<REGION>
+PR_KEY_PW=<PASSPHRASE>
+BUCKET=<BUCKET>
+ISO_NAME=<ISO_NAME>
+OS="<OS_NAME>"
+RU="<RHEL_UNAME>"
+RP="<RHEL_PASS>"
+SAC_OCID=<ZEROS_OCID>
+
+LOCAL_IP=`curl -s http://169.254.169.254/opc/v1/vnics/0/privateIp`
+AD=`curl -s http://169.254.169.254/opc/v1/instance/availabilityDomain`
+COMPARTMENT=`curl -s http://169.254.169.254/opc/v1/instance/compartmentId`
+VNIC_ID=`curl -s http://169.254.169.254/opc/v1/vnics/0/vnicId`
+CIDR_BLOCK=`curl -s http://169.254.169.254/opc/v1/vnics/0/subnetCidrBlock`
+MY_OCID=`curl -s http://169.254.169.254/opc/v1/instance/id`
+
+### DO NOT CHANGE THIS SHAPE!!! ###
+SHAPE="BM.Standard1.36"
+
+yum -y -q update
+yum -y -q install sharutils
+yum -y -q install httpd rsync lzma gcc libffi-devel python-devel openssl-devel unzip gcc rsyslog jq python-firewall
+
+sed -i.bak 's/<\/zone>/  <service name=\"http\"\/>/' /etc/firewalld/zones/public.xml
+echo "</zone>" >> /etc/firewalld/zones/public.xml
+restorecon -Rv /etc/firewalld/zones/public.xml
+service firewalld restart
+
+systemctl enable httpd
+systemctl start httpd
+
+mkdir /var/www/isos
+mkdir /mnt/iso
+mkdir /var/www/html/ipxe
+mkdir /var/www/html/oci
+mkdir /var/www/html/<OS_NAME>
+mkdir -p /home/opc/.oci
+
+chmod 755 /home/opc/.oci
+chmod 777 /var/www/isos
+
+easy_install pip
+pip install pip --upgrade
+pip install oci-cli
+
+sed '/imtcp/s/#//g' -i.bak /etc/rsyslog.conf
+sed '/InputTCPServerRun/s/#//g' -i.bak /etc/rsyslog.conf
+systemctl restart rsyslog.service
+
+cd /home/opc/.oci
+if [ -z "${PR_KEY_PW}" ]
+then
+	cat > config <<-EOF
+[DEFAULT]
+user=${USER}
+fingerprint=${FINGERPRINT}
+key_file=/home/opc/.oci/oci_api_key.pem
+tenancy=${TENANCY}
+region=${REGION}
+EOF
+else
+	cat > config <<-EOF
+[DEFAULT]
+user=${USER}
+fingerprint=${FINGERPRINT}
+key_file=/home/opc/.oci/oci_api_key.pem
+tenancy=${TENANCY}
+region=${REGION}
+pass_phrase=${PR_KEY_PW}
```

Comment:
Not 100% sure, but I believe that if there is no passphrase then you can just set "pass_phrase=" and it will be fine. So if you're looking to reduce the character count, that's something you can try.

Reply:
Yes, but if a pass_phrase is set, I need this in the config file that is generated as part of this.  This segment generates two different config files based on whether the passphrase is set.  I have tried setting the pass_phrase to null in the past and received errors.  If this behavior has changed I am unaware.